src/nbl/video/COpenGLBuffer.cpp

```diff
@@ -8,9 +8,12 @@ namespace nbl {
 namespace video
 {
 
+std::atomic_uint32_t COpenGLBuffer::s_reallocCounter = 0u;
+
 COpenGLBuffer::~COpenGLBuffer()
 {
-    m_device->destroyBuffer(BufferName);
+    auto* device = static_cast<IOpenGL_LogicalDevice*>(const_cast<ILogicalDevice*>(getOriginDevice()));
```

Comment:
you probably want to make `getOriginDevice` a protected template function and have it do all the const_casting and static_casting

Reply:
TODO